### PR TITLE
blog on comparing sorting algorithms

### LINKED ISSUE #1940: blog on comparing sorting algorithms

### PR DESCRIPTION
**Pull request by jainaryan04**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1940 

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: blog/comparing-sortings.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+# Comparison of Sorting Algorithms
+
+Sorting algorithms are essential for organizing data, making it easier to analyze, search, and manipulate. Here’s an overview of some commonly used sorting algorithms, comparing their time complexity, efficiency, and applications.
+
+## 1. Bubble Sort
+**Description**: Bubble Sort repeatedly steps through the list, compares adjacent elements, and swaps them if needed until the list is sorted.
+
+- **Time Complexity**: \(O(n^2)\) average and worst-case; \(O(n)\) best-case (already sorted)
+- **Space Complexity**: \(O(1)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: replace `\(O(n^2)\)` with `$O(n^2)$`, `\(O(n)\)` with `$O(n)$` , and `\(O(1)\)` with `$\(O(1)\)$`

File: blog/comparing-sortings.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+- **Pros**: Simple and easy to implement
+- **Cons**: Inefficient for large datasets
+- **Use Case**: Suitable for small or nearly sorted datasets
+
+## 2. Selection Sort
+**Description**: Selection Sort divides the list into sorted and unsorted parts, finding the minimum element in the unsorted part and placing it in order.
+
+- **Time Complexity**: \(O(n^2)\) for all cases
+- **Space Complexity**: \(O(1)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- **Pros**: Simple implementation; fewer swaps
+- **Cons**: Inefficient for large datasets
+- **Use Case**: Useful for small datasets or when memory writes are costly
+
+## 3. Insertion Sort
+**Description**: Insertion Sort builds a sorted list by taking one item from the unsorted portion and inserting it into the correct position.
+
+- **Time Complexity**: \(O(n^2)\) average and worst-case; \(O(n)\) best-case (already sorted)
+- **Space Complexity**: \(O(1)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+- **Pros**: Efficient for small or nearly sorted lists
+- **Cons**: Not suitable for large unsorted datasets
+- **Use Case**: Good for small or mostly sorted datasets
+
+## 4. Merge Sort
+**Description**: Merge Sort is a divide-and-conquer algorithm that splits the list into halves, sorts each half, and then merges them.
+
+- **Time Complexity**: \(O(n \log n)\) for all cases
+- **Space Complexity**: \(O(n)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+- **Cons**: Requires extra space
+- **Use Case**: Effective for large datasets and linked lists
+
+## 5. Quick Sort
+**Description**: Quick Sort is a divide-and-conquer algorithm that selects a "pivot" element, partitions the array around it, and recursively sorts partitions.
+
+- **Time Complexity**: \(O(n \log n)\) average; \(O(n^2)\) worst-case
+- **Space Complexity**: \(O(\log n)\)
+- **Pros**: Fast in practice due to in-place sorting
+- **Cons**: Worst-case time can be \(O(n^2)\); not stable
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+- **Use Case**: Suitable for large datasets, especially when optimizing for speed
+
+## 6. Heap Sort
+**Description**: Heap Sort uses a binary heap to sort by repeatedly extracting the largest element (root) and adjusting the heap.
+
+- **Time Complexity**: \(O(n \log n)\)
+- **Space Complexity**: \(O(1)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+- **Pros**: Efficient, in-place sorting
+- **Cons**: Not stable; slower in practice than Quick Sort
+- **Use Case**: Used when in-place sorting is preferred and stability isn't essential
+
+## 7. Radix Sort
+**Description**: Radix Sort is a non-comparative algorithm that sorts integers by processing digits, usually using Counting Sort as a subroutine.
+
+- **Time Complexity**: \(O(d \cdot (n + k))\), where \(d\) is the digit count and \(k\) is the digit range
+- **Space Complexity**: \(O(n + k)\)
+- **Pros**: Fast for fixed-digit data; stable
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+- **Cons**: Limited to numbers or strings; extra space needed
+- **Use Case**: Ideal for sorting large lists of integers or strings of uniform length
+
+## 8. Counting Sort
+**Description**: Counting Sort counts occurrences of each element and calculates their sorted positions.
+
+- **Time Complexity**: \(O(n + k)\), where \(k\) is the input range
+- **Space Complexity**: \(O(n + k)\)
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

File: blog/comparing-sortings.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+| Bubble Sort     | \(O(n)\)             | \(O(n^2)\)            | \(O(1)\)         | Yes    | Small or mostly sorted datasets         |
+| Selection Sort  | \(O(n^2)\)           | \(O(n^2)\)            | \(O(1)\)         | No     | Small datasets, costly memory writes    |
+| Insertion Sort  | \(O(n)\)             | \(O(n^2)\)            | \(O(1)\)         | Yes    | Small or nearly sorted datasets         |
+| Merge Sort      | \(O(n \log n)\)      | \(O(n \log n)\)       | \(O(n)\)         | Yes    | Large datasets, stable sorting needed   |
+| Quick Sort      | \(O(n \log n)\)      | \(O(n^2)\)            | \(O(\log n)\)    | No     | Large datasets, in-place and fast sort  |
+| Heap Sort       | \(O(n \log n)\)      | \(O(n \log n)\)       | \(O(1)\)         | No     | Large datasets, stability not required  |
+| Radix Sort      | \(O(n \cdot d)\)     | \(O(n \cdot d)\)      | \(O(n + k)\)     | Yes    | Large lists of numbers, fixed-length    |
+| Counting Sort   | \(O(n + k)\)         | \(O(n + k)\)          | \(O(n + k)\)     | Yes    | Small integer range, limited data range |
```
Comment (ajay-dhangar) [inline blog/comparing-sortings.md]: use `$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by jainaryan04** [comment]
done @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by ajay-dhangar** [comment]
To add yourself as an author (`ARYAN-JAIN`), please add your information to the following file: https://github.com/ajay-dhangar/algo/blob/main/blog/authors.yml. Please use the following format: 

```yml
narendra-dhangar:
  name: Narendra Dhangar
  title: B.Tech (CSE) Student
  url: https://www.instagram.com/narendra_dh._05/
  image_url: https://github.com/narendra-dhangar.png
  email: narendradhangar63@gmail.com
  page: true # Turns the feature on
  description: >
    A dedicated Computer Science and Engineering student with a passion for problem-solving and algorithms. I am constantly exploring various fields of computer science, from data structures and algorithms to web development and machine learning. As an avid learner, I enjoy tackling challenging problems and sharing knowledge with the community. Eager to contribute to open-source projects and make an impact in the tech world.

  socials:
    linkedin: narendra-dhangar-880558297
    github: narendra-dhangar
```

**Comment by jainaryan04** [comment]
@ajay-dhangar I am already in the file

**Comment by jainaryan04** [comment]
@ajay-dhangar please merge

**Review by ajay-dhangar — APPROVED** [review]
Your mistake is that you use the author as "ARYAN-JAIN", but in the author file the correct name is "Aryan-Jain". So I fix this. And Done

**Pull request merged**